packages/mattermost-redux/src/selectors/entities/threads.ts

```diff
@@ -66,16 +67,24 @@ export function getThreads(state: GlobalState): IDMappedObjects<UserThread> {
     return state.entities.threads.threads;
 }
 
-export function getThread(state: GlobalState, threadId: $ID<UserThread> | undefined): UserThread | null {
-    if (
-        !threadId ||
-        !(getThreadsInCurrentTeam(state)?.includes(threadId) || getUnreadThreadsInCurrentTeam(state)?.includes(threadId))
-    ) {
-        return null;
-    }
+export const getThread: (state: GlobalState, threadId: $ID<UserThread> | undefined) => UserThread | null = createSelector(
+    'getThread',
+    getThreads,
+    (state: GlobalState) => getMyChannels(state).map((c) => c.id),
```

Comment:
This would also need its own memoization since it's returning a new array of channel IDs each time this is called, even if the state hasn't changed.